### PR TITLE
[FIX] stock: initial demand and immediate transfer

### PR DESCRIPTION
**Pull request by sle-odoo**
Create an immediate transfer for a receipt, add a move, set a quantity
done and due to rev[0], an initial demand is set, then
_autoconfirm_picking is called and action_confirm of picking calls
_action_assign if the location source bypasses the reservation. Set
again an inferior qty_done, the initial demand is also updated due to
rev[0] and somehow the system tries to write on a now unlinked move
line.

fixes
- never reserve an immediate transfer
- never update the initial demand if the move is reserved to fix the
existing databases.

[0] 8303b1a69e96807bcb548ef079dcb463e34df36b

opw-2180556

### CODE REVIEW

**Comment by sle-odoo** [comment]
cc @fah-odoo

File: addons/stock/models/stock_picking.py [hunk 1]
```diff
@@ -621,7 +621,7 @@ def action_confirm(self):
             .filtered(lambda move: move.state == 'draft')\
             ._action_confirm()
         # call `_action_assign` on every confirmed move which location_id bypasses the reservation
-        self.filtered(lambda picking: picking.location_id.usage in ('supplier', 'inventory', 'production') and picking.state == 'confirmed')\
+        self.filtered(lambda picking: not picking.immediate_transfer and picking.location_id.usage in ('supplier', 'inventory', 'production') and picking.state == 'confirmed')\
```
Comment (amoyaux) [inline addons/stock/models/stock_picking.py]: You probably need to add an immediate_transfer check on _compute_state

**Comment by sle-odoo** [comment]
@amoyaux ready to review

**Comment by amoyaux** [comment]
robodoo r+ rebase-ff

**Pull request closed without merging**